backend/grpc.go

```diff
@@ -0,0 +1,46 @@
+package backend
+
+import (
+	"context"
+
+	bproto "github.com/grafana/grafana-plugin-sdk-go/genproto/go/grafana_plugin"
+)
+
+type GRPCClient struct {
+	client bproto.BackendPluginClient
+}
+
+// BackendPlugin is the Grafana Backend plugin interface.
+// It corresponds to: grafana.plugin protobuf: BackendPlugin Service | genproto/go/grafana_plugin: BackendPluginClient interface
+type BackendPlugin interface {
+	Check(ctx context.Context, req *bproto.PluginStatusRequest) (*bproto.PluginStatusResponse, error)
+	DataQuery(ctx context.Context, req *bproto.DataQueryRequest) (*bproto.DataQueryResponse, error)
+}
+
+type grpcServer struct {
+	Impl backendPluginWrapper
+}
+
+func (m *GRPCClient) DataQuery(ctx context.Context, req *bproto.DataQueryRequest) (*bproto.DataQueryResponse, error) {
+	return m.client.DataQuery(ctx, req)
+}
+
+func (m *grpcServer) DataQuery(ctx context.Context, req *bproto.DataQueryRequest) (*bproto.DataQueryResponse, error) {
+	return m.Impl.DataQuery(ctx, req)
+}
+
+func (m *GRPCClient) Check(ctx context.Context, req *bproto.PluginStatusRequest) (*bproto.PluginStatusResponse, error) {
+	return m.client.Check(ctx, req)
+}
+
+func (m *grpcServer) Check(ctx context.Context, req *bproto.PluginStatusRequest) (*bproto.PluginStatusResponse, error) {
+	return m.Impl.Check(ctx, req)
+}
+
+func (m *GRPCClient) REST(ctx context.Context, req *bproto.RESTRequest) (*bproto.RESTResponse, error) {
```

Comment:
I'm not wild about using `REST` as the actual function name -- but I struggled with alternatives.  While desiging the proposal, I went though `RESTish`, `HTTPish`, and finally just `PluginRequest` -- that last was @bergquist suggestion since REST and HTTP are more inspiration for how it can be used.

Reply:
Maybe "resource" is an OK name?  essentially HTTP style verbs on a resource path?

Reply:
I'd be okay with Resource if we don't think of something better. I don't like REST either, but since all Requests are Plugin Requests, I didn't really like that name.